uniffi/src/scaffolding.rs

```diff
@@ -77,7 +77,7 @@ mod filters {
         // By explicitly naming the type here, we help the rust compiler to type-check the user-provided
         // implementations of the functions that we're wrapping (and also to type-check our generated code).
         Ok(format!(
-            "<{} as uniffi::support::ViaFfi>::into_ffi_value({})",
+            "<{} as uniffi::support::ViaFfi>::into_ffi_value(&{})",
```

Comment:
This seems OK to me, but I do wonder if there are cases where we actually *want* to take an owned reference, e.g. so we can transfer ownership to the foreign language side without a copy. That can be for follow-up work though.